lib/client.js

```diff
@@ -62,21 +62,9 @@ class Parser extends HTTPParser {
 
     if (body) {
       if (typeof body.write === 'function') {
-        // TODO: 'error' and 'close' should only resume if waiting for 'drain'.
-        // TODO: Don't assume 'error' and 'close' will set destroyed to ´true`.
         body
           .once('drain', this.resume)
-          .once('close', this.resume)
-        if (EventEmitter.errorMonitor) {
-          body.once(EventEmitter.errorMonitor, this.resume)
-        } else {
-          body.once('error', (err) => {
-            this.resume()
-            if (body.listenerCount('error') === 0) {
-              body.emit('error', err)
-            }
-          })
-        }
+          .once('error', this.resume)
```

Comment:
We will always either emit `'error'` unless gracefully ended. That's the only case we need resume.